presto-main/src/main/java/io/prestosql/operator/DynamicFilterCollectOperator.java

```diff
@@ -0,0 +1,207 @@
+ * The collected pages' value are used for creating a run-time filtering constraint (for probe-side table scan in an inner join).
+ * We support only small build-side pages (which should be the case when using "broadcast" join).
+ */
+public class DynamicFilterCollectOperator
+        implements Operator
+{
+    public static class DynamicFilterCollectOperatorFactory
+            implements OperatorFactory
+    {
+        private final int operatorId;
+        private final PlanNodeId planNodeId;
+        private final LocalDynamicFilter dynamicFilter;
+        private final TypeProvider types;
+
+        public DynamicFilterCollectOperatorFactory(int operatorId, PlanNodeId planNodeId, LocalDynamicFilter dynamicFilter, TypeProvider types)
+        {
+            this.operatorId = operatorId;
+            this.planNodeId = requireNonNull(planNodeId, "planNodeId is null");
+            this.dynamicFilter = requireNonNull(dynamicFilter, "dynamicFilter is null");
+            this.types = requireNonNull(types, "types is null");
+        }
+
+        @Override
+        public Operator createOperator(DriverContext driverContext)
+        {
+            return new DynamicFilterCollectOperator(
+                    driverContext.addOperatorContext(operatorId, planNodeId, DynamicFilterCollectOperator.class.getSimpleName()),
+                    dynamicFilter,
+                    types);
+        }
+
+        @Override
+        public void noMoreOperators()
+        {
+        }
+
+        @Override
+        public OperatorFactory duplicate()
+        {
+            throw new UnsupportedOperationException("duplicate() is not supported for DynamicFilterCollectOperatorFactory");
+        }
+    }
+
+    private final OperatorContext context;
+    private boolean finished;
+    private Page current;
+    private final Consumer<TupleDomain<DynamicFilters.Descriptor>> dynamicPredicateConsumer;
+    private final int maxValueCount;
+    private final long maxByteSize;
+
+    private final List<LocalDynamicFilter.Channel> channels;
+    private final Type[] buildTypes;
+    private HashSet<Object>[] valueSets;  // may be dropped if the predicate becomes too large.
+    private boolean[] nullAllowed;
+
+    private long valuesCount;
+    private long sizeInBytes;
+
+    public DynamicFilterCollectOperator(OperatorContext context, LocalDynamicFilter dynamicFilter, TypeProvider types)
+    {
+        this.context = requireNonNull(context, "context is null");
+        this.maxValueCount = getDynamicFilteringMaxPerDriverValueCount(context.getSession());
+        this.maxByteSize = getDynamicFilteringMaxPerDriverByteSize(context.getSession()).toBytes();
+
+        this.dynamicPredicateConsumer = dynamicFilter.getTupleDomainConsumer();
+        this.channels = dynamicFilter
+                .getChannels()
+                .stream()
+                .filter(channel -> channel.getDescriptor().getInput() instanceof SymbolReference)
```

Comment:
That line
`channel.getDescriptor().getInput() instanceof SymbolReference`
doesn't seem to be valid.
`Descrptor` is probe side specific. Why would `DynamicFilterCollectOperator` care if it's a symbol reference or not?

Reply:
Good point, thanks! 
Changed `DynamicFilterCollectOperator` so it won't have to work with `Descriptor`s.